bin/git-forgit

```diff
@@ -1176,6 +1189,141 @@ _forgit_paths_list() {
     find "$path" -name "*$ext" -print |sed -e "s#$ext\$##" -e 's#.*/##' -e '/^$/d' | sort -fu
 }
 
+_forgit_filter_existing_paths() {
+    while read -r path; do
+        [[ -d "$path" ]] && echo "$path"
+    done
+}
+
+_forgit_worktree_preview() {
+    local sha
+    # trailing space in grep to avoid matching worktrees with a common path
+    sha=$(git worktree list | grep "$1 " | awk '{print $2}')
+    if [[ "$sha" == "(bare)" ]]; then
+        printf "%b(bare)%b %s\n" '\e[0;33m' '\e[0m' 'No history for git dir'
+        return
+    fi
+    # the trailing '--' ensures that this works for branches that have a name
+    # that is identical to a file
+    git log "$sha" "${_forgit_log_preview_options[@]}" --
+}
+
+_forgit_worktree_select() {
+    _forgit_inside_work_tree || _forgit_inside_git_dir || return 1
+    local worktree_list count tree opts
+    worktree_list=$(git worktree list | grep -vE "prunable$" | awk '{print $1}' | _forgit_filter_existing_paths)
+
+    count=$(echo "$worktree_list" | wc -l)
+    [[ $count -eq 1 ]] && return 1
+
+    opts="
+        $FORGIT_FZF_DEFAULT_OPTS
+        +s +m --tiebreak=index
+        --preview=\"$FORGIT worktree_preview {1}\"
+        $FORGIT_WORKTREE_SELECT_FZF_OPTS
+    "
+
+    tree=$(echo "$worktree_list" | FZF_DEFAULT_OPTS="$opts" fzf)
+    [[ -z "$tree" ]] && return 1
+    echo "$tree"
+}
+
+_forgit_worktree_lock_max_args() {
+    local usage
+    usage=$(git worktree lock --help 2>/dev/null | grep -E 'usage: git worktree lock' | head -n1)
+    usage=${usage#*:}
+     rest=$(echo "$usage" | awk '{for(i=4;i<=NF;++i)printf "%s ",$i; print ""}')
+    # shellcheck disable=SC2206
+    local args=($rest)
+    echo "${#args[@]}"
+}
+
+_forgit_git_worktree_lock() {
+    _forgit_worktree_lock_git_opts=()
+    _forgit_parse_array _forgit_worktree_lock_git_opts "$FORGIT_WORKTREE_LOCK_GIT_OPTS"
+    git worktree lock "${_forgit_worktree_lock_git_opts[@]}" "$@"
+}
```

Comment:
Would be nice to allow operating on multiple worktrees at the same time. Should be easy to implement by allowing multiselect in fzf and modifiying the `_forgit_git_worktree_*` functions like this:

```suggestion
_forgit_git_worktree_lock() {
    local trees
    trees=$1
    shift
    _forgit_worktree_lock_git_opts=()
    _forgit_parse_array _forgit_worktree_lock_git_opts "$FORGIT_WORKTREE_LOCK_GIT_OPTS"
    echo "$trees" | xargs -I% git worktree lock "${_forgit_worktree_lock_git_opts[@]}" %
}
```